src/upgrade.c

```diff
@@ -158,6 +164,16 @@ static int upgrade_main(void *ctx, struct apk_database *db, struct apk_string_ar
 	if (uctx->self_upgrade_only)
 		return 0;
 
+  if (uctx->ignore) {
```

Comment:
If use the arguments for ignoring certain packages, I think the opposite should work too. That is calling `upgrade` without the `--ignore` flag but a list of package names, should upgrade those packages only. Though, this might be worth a separate comment.

Reply:
I think I will open a new issue in order to keep the two features separated. I'm also really interested in this new one since I already had some usecases where I wanted to upgrade only one package and ended up with a full system upgrade.

If it's ok for you, I will work on this after the current feature.